src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol

```diff
@@ -0,0 +1,451 @@
+        return reserveRatioForBuying;
+    }
+
+    /// @inheritdoc IBancorVirtualSupplyBondingCurveFundingManager
+    function getReserveRatioForSelling() external view returns (uint32) {
+        return reserveRatioForBuying;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    /// @inheritdoc IFundingManager
+    function token() public view returns (IERC20) {
+        return __Module_orchestrator.token();
+    }
+
+    //--------------------------------------------------------------------------
+    // OnlyOrchestrator Functions
+
+    /// @inheritdoc IBancorVirtualSupplyBondingCurveFundingManager
+    function mintIssuanceTokenTo(address _receiver, uint _amount)
+        external
+        onlyOrchestratorOwner
+        validReceiver(_receiver)
+    {
+        _mint(_receiver, _amount);
+    }
+
+    /// @inheritdoc IVirtualTokenSupply
+    function setVirtualTokenSupply(uint _virtualSupply)
+        external
+        override(VirtualTokenSupplyBase)
+        onlyOrchestratorOwner
+    {
+        _setVirtualTokenSupply(_virtualSupply);
+    }
+
+    /// @inheritdoc IVirtualCollateralSupply
+    function setVirtualCollateralSupply(uint _virtualSupply)
+        external
+        override(VirtualCollateralSupplyBase)
+        onlyOrchestratorOwner
+    {
+        _setVirtualCollateralSupply(_virtualSupply);
+    }
+
+    /// @inheritdoc IBancorVirtualSupplyBondingCurveFundingManager
+    function setReserveRatioForBuying(uint32 _reserveRatio)
+        external
+        onlyOrchestratorOwner
+    {
+        _setReserveRatioForBuying(_reserveRatio);
+    }
+
+    /// @inheritdoc IBancorVirtualSupplyBondingCurveFundingManager
+    function setReserveRatioForSelling(uint32 _reserveRatio)
+        external
+        onlyOrchestratorOwner
+    {
+        _setReserveRatioForSelling(_reserveRatio);
+    }
+
+    //--------------------------------------------------------------------------
+    // Upstream Function Implementations
+
+    /// @dev Calculates the amount of tokens to mint for a given deposit amount using the Bancor formula.
+    /// This internal function is an override of BondingCurveFundingManagerBase's abstract function.
+    /// It handles decimal conversions and calculations through the bonding curve.
+    /// @param _depositAmount The amount of collateral deposited to purchase tokens.
+    /// @return mintAmount The amount of tokens that will be minted.
+    function _issueTokensFormulaWrapper(uint _depositAmount)
+        internal
+        view
+        override(BondingCurveFundingManagerBase)
+        returns (uint mintAmount)
+    {
+        // Convert depositAmount to 18 decimals, which is required by Bancor formula
+        uint decimalConvertedDepositAmount = _convertAmountToRequiredDecimal(
+            _depositAmount,
+            IERC20MetadataUpgradeable(address(token())).decimals(),
```

Comment:
Do we want this kind of gas optimization? Because it seems that we possibly start duplicating information/variables a lot. Any thoughts? @0xNuggan @saxenism @marvinkruse

Reply:
indeed you should not duplicate information or variables if possible, but you could use constnats instead of function calls. 
E.g. can just write `18`, or have a `constants.sol` file that you import in which you define a `TOKEN_DECIMALS = 18`

Reply:
I think we can store the decimals on initialization. The issue/redeem token functions are probably going to be among the most-called functions of the module...